generator/dryrun.py

```diff
@@ -70,13 +69,17 @@ def __init__(
         """Initialize dry run instance."""
         self.sql = sql
         self.use_cloud_function = use_cloud_function
-        self.client = client
         self.project = project
         self.dataset = dataset
         self.table = table
         self.dry_run_url = dry_run_url
         self.id_token = id_token
 
+    @cached_property
+    def client(self):
+        """Get BigQuery client instance."""
+        return bigquery.Client()
+
```

Comment:
moving BigQuery client instance creation into the class as it's not possible to pass a client instance across different processes

Reply:
> Using the cloud function, generation completes in about 5 minutes. Using service credentials, generation takes about 10 minutes as the Bigquery client needs to be instantiated for every process as it cannot be shared across processes.

It might be worth trying to initialize one set of BigQuery credentials (which do appear to support pickling) and share those across the processes (passing them in to the `bigquery.Client` constructor's `credentials` argument) to see if that can speed things up closer to the cloud function performance, as [Google's docs](https://cloud.google.com/apis/docs/client-libraries-best-practices#reuse_client_objects_and_sessions) indicate doing the authentication, authorization, and access token generation can take multiple seconds.